### PR TITLE
Remove default props from func components

### PR DESCRIPTION
**Pull request by DevStarks**
Remove deprecated `defaultProps` on function components

https://github.com/facebook/react/pull/16210

### CODE REVIEW

File: src/CopyToClipboardButton/CopyToClipboardButton.tsx [hunk 1]
```diff
@@ -72,10 +72,4 @@ function CopyToClipboardButton({
   );
 }
 
-CopyToClipboardButton.defaultProps = {
-  copyText: '',
-  displayText: undefined,
-  variant: ButtonVariants.NEUTRAL,
-};
```
Comment (DevStarks) [inline src/CopyToClipboardButton/CopyToClipboardButton.tsx:79]: These were already added to prop object property defaults above

File: src/Drawer/DrawerHeader.tsx [hunk 2]
```diff
@@ -58,9 +58,4 @@ function DrawerHeader({
   );
 }
 
-DrawerHeader.defaultProps = {
-  bordered: true,
-  title: null,
-};
```
Comment (DevStarks) [inline src/Drawer/DrawerHeader.tsx:64]: - `bordered` is already declared above
- `title` I switched to `undefined` based on usage

File: src/Dropdown/DropdownDivider.tsx [hunk 3]
```diff
@@ -34,10 +34,4 @@ function DropdownDivider({
   );
 }
 
-DropdownDivider.defaultProps = {
-  as: 'hr',
-  bsPrefix: 'dropdown',
```
Comment (DevStarks) [inline src/Dropdown/DropdownDivider.tsx:39]: Declared above

**Review by kyleshike — APPROVED** [review]
NICE

**Pull request merged**